R/Rules-methods.R

```diff
@@ -555,15 +555,81 @@ setMethod(
     data = "Data"
   ),
   definition = function(nextBest, doselimit = Inf, samples, model, data, ...) {
-    modelfit <- fit(samples, model, data)
-    doses <- modelfit$dose
-    is_dose_eligible <- doses <= doselimit
-    doses_eligible <- doses[is_dose_eligible]
-    dlt_prob <- modelfit$middle[is_dose_eligible]
-    next_dose_level <- which.min(abs(dlt_prob - nextBest@target))
-    next_dose <- doses_eligible[next_dose_level]
 
-    list(value = next_dose)
+    # Matrix with samples from the dose-tox curve at the dose grid points.
+    prob_samples <- sapply(data@doseGrid, prob, model = model, samples = samples)
+    dlt_prob <- colMeans(prob_samples)
+
+    # Determine the dose with the closest distance
+    dose_target <- data@doseGrid[which.min(abs(dlt_prob - nextBest@target))]
+
+    # Determine next dose
+    doses_eligible <- h_next_best_eligible_doses(
+      data@doseGrid,
+      doselimit,
+      data@placebo
+    )
+    next_dose_level_eligible <- which.min(abs(doses_eligible - dose_target))
+    next_dose <- doses_eligible[next_dose_level_eligible]
```

Comment:
yeah maybe the logic can be simplified a bit, or made more robust. in general we can build on the assumptions you mention Wojciech, but yeah maybe somebody comes up with a dose-tox model that is one-off and not monotonic

Reply:
Thanks @0liver0815 - I think there are some models that the current order is justified and therefore correct (like those for which we have many samples for a given level of dose and then we take some dose from the density). But in this case I think it would be good to correct it now as the correction is quite simple. But, I think that we would need to quickly revisit all the models against this.

Reply:
Problem is that model is separated from nextbest rule. So I would say let's make a separate issue for this (check nextbest models for non Monotonic model behavior) and continue here as planned

Reply:
Thanks @danielinteractive Just created, it is: https://github.com/Roche/crmPack/issues/488